src/scportrait/pipeline/extraction.py

```diff
@@ -524,206 +509,145 @@ def _extract_classes(self, px_center, arg, return_failed_ids=False):
             ids.append(cytosol_id)
 
         # get region that should be extracted
-        _px_center = px_center[index]
-        window_y = slice(_px_center[1] - self.width_extraction, _px_center[1] + self.width_extraction)
-        window_x = slice(_px_center[0] - self.width_extraction, _px_center[0] + self.width_extraction)
+        window_y = slice(px_center[1] - self.width_extraction, px_center[1] + self.width_extraction)
+        window_x = slice(px_center[0] - self.width_extraction, px_center[0] + self.width_extraction)
 
         # ensure that the cell is not too close to the image edge to be extracted
         condition = [
-            self.width_extraction < _px_center[0],
-            _px_center[0] < self.input_image_width - self.width_extraction,
-            self.width_extraction < _px_center[1],
-            _px_center[1] < self.input_image_height - self.width_extraction,
+            self.width_extraction < px_center[0],
+            px_center[0] < self.input_image_width - self.width_extraction,
+            self.width_extraction < px_center[1],
+            px_center[1] < self.input_image_height - self.width_extraction,
         ]
-        if np.all(condition):
-            extraction_status = True
-        else:
-            extraction_status = False
-
-        if extraction_status:
-            masks = []
-
-            # get the segmentation masks
-            for mask_ix in range(self.n_masks):
-                if image_index is None:
-                    # nuclei_mask = self.sdata[self.nucleus_key].data[window_y, window_x].compute()
-                    mask = self.seg_masks[mask_ix, window_y, window_x]
-                else:
-                    # nuclei_mask = self.sdata[self.nucleus_key].data[image_index, window_y, window_x].compute()
-                    mask = self.seg_masks[image_index, mask_ix, window_y, window_x]
-
-                # modify nucleus mask to only contain the nucleus of interest and perform some morphological operations
-                mask = np.where(mask == ids[mask_ix], 1, 0)
-                mask = binary_fill_holes(mask)
-                mask = gaussian(mask, preserve_range=True, sigma=1)
+        assert np.all(condition), "Cell is too close to the image edge to be extracted."
```

Comment:
```suggestion
        if not condition:
            msg = "Cell is too close to the image edge to be extracted."
            raise ValueError(msg) # or a custom error
```
Something like this - `assert` can be optimized out of code so should only really be used for developers https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement